streams/src/test/java/org/apache/kafka/streams/integration/StoreQueryIntegrationTest.java

```diff
@@ -477,6 +477,7 @@ public void shouldQueryStoresAfterAddingAndRemovingStreamThread() throws Excepti
         //Add thread
         final Optional<String> streamThread = kafkaStreams1.addStreamThread();
         assertThat(streamThread.isPresent(), is(true));
+        until(() -> kafkaStreams1.state().isRunningOrRebalancing());
```

Comment:
It seems like this isn't used. The rebalancing will end up in a running state which is waited for below. I suggest only have the condition below waiting for a running state before query the state store.

Reply:
You are right. I see the point that it should not make any functional difference.

According to `KafkaStreams.State`: 

> REBALANCING state will transit to RUNNING if all of its threads are in RUNNING state

( so as soon as the new `StreamThread` is in `RUNNING`)

Although `REBALANCING` is explicitly stated in [KAFKA-13246](https://issues.apache.org/jira/browse/KAFKA-13246):

> StoreQueryIntegrationTest#shouldQueryStoresAfterAddingAndRemovingStreamThread should be improved by waiting for the client to go to rebalancing or running after adding and removing a thread. It should also wait until running before querying the state store 

One small benefit I see here is to assert in isolation of any following code that adding/removing a thread was indeed successful.

@wcarlson5 maybe you as the author of the JIRA issue could give us a hint if we are overlooking anything here ?
Thanks

Reply:
It is true that we are waiting for state below, however the way we are doing so causes some exceptions due to streams being in the wrong state. These exceptions are really not the point of any of the tests in this file. So how we have gotten around that was white listing exception messages. It would be better just to wait for the correct state before trying do do any IQ, like what is done here. @andy0x01 the isolation of the add remove is a nice benefit as well.